vsintegration/src/FSharp.Editor/CodeFixes/FSharpCodeFixContext.fs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+
+open CancellableTasks
+
+type FSharpCodeFixContext(document: Document, span: TextSpan) =
+
+    let sourceText =
+        lazy
+            (cancellableTask {
+                let! cancellationToken = CancellableTask.getCurrentCancellationToken ()
+                return! (document.GetTextAsync cancellationToken)
```

Comment:
document.GetTextAsync already has it's own cache AFAIK, it should not do an expensive op for a second time.
 ( member > let binding > lazy  > cancellableTask > document) feels like too much of an indirection, each keeping it's own reference to something.

Reply:
Yep, didn't think about it, removing the lazy then.